app/controllers/sessions_controller.rb

```diff
@@ -4,22 +4,14 @@ def new
   end
 
   def create
-    if request.env['omniauth.auth']
-      user = User.create_with_omniauth(request.env['omniauth.auth'])
+    user = User.find_by(email: params[:email])
+    if user && user.authenticate(params[:password])
```

Comment:
I moved all the oauth stuff to a different controller so I could better visualize what was going on.

Reply:
Love that abstraction! (encapsulation?)

Reply:
Nice, this looks a lot cleaner. Way to keep those controller SKINNY.

Reply:
This will also be a great separation if we ever add in google oath.

Reply:
Great work here!